src/ext/GeometryUtil.js

```diff
@@ -55,55 +82,67 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
 	// Converts a metric distance to one of [ feet, nauticalMile, metric or yards ] string
 	//
 	// @alternative
-	// @method readableDistance(distance, isMetric, useFeet, isNauticalMile): string
+	// @method readableDistance(distance, isMetric, useFeet, isNauticalMile, precision): string
 	// Converts metric distance to distance string.
-	readableDistance: function (distance, isMetric, isFeet, isNauticalMile) {
+	// The value will be rounded as defined by the precision option object.
+	readableDistance: function (distance, isMetric, isFeet, isNauticalMile, precision) {
 		var distanceStr,
-			units;
+			units,
+			precision = L.Util.extend({}, defaultPrecision, precision);
 
-		if (typeof isMetric == "string") {
-			units = isMetric;
+		if (isMetric) {
+			units = typeof isMetric == 'string' ? isMetric : 'metric';
```

Comment:
Hi,

This would already fix issue #690.

See also https://stackoverflow.com/questions/43981845/how-can-i-change-my-leaflet-draw-units-from-imperial-to-metric

The bug is not caught by tests because the latter only test `L.GeometryUtil.readableDistance` directly, by calling it with only 2 arguments (`distance` and `isMetric`), whereas in the context of `L.Draw.Polyline`, it is also called with the `isFeet` 3rd argument, which is `true` by default.

If this PR is too big to be merged soon, maybe a simple one with just that fix could be sent in the meantime?